src/pages/Recipe.js

```diff
@@ -35,37 +94,189 @@ export default class Recipe extends Component {
       recipe.flavorTotal += flavor.percent;
     });
 
+    if (recipe.notes) {
+      const newNotes = recipe.notes
+        .toString()
+        .split('\n')
+        .map((line, i) => {
+          return <p key={i}>{line}</p>;
+        });
+
+      recipe.notes = newNotes;
+    }
+
+    const newTags = recipe.tags.map((tag, i) => {
+      return (
+        <a className="link--tags" key={i} href="/">
+          {tag}
+        </a>
+      );
+    });
+
+    recipe.tags = newTags;
+
+    this.setState(recipe);
+  }
+
+  compareWithStash() {
+    // eslint-disable-next-line
+    console.log(this.state);
+    const query = this.props.location.search;
+    const queryValues = queryString.parse(query);
+    const pageId = parseFloat(queryValues.id);
+
+    const recipe = recipes.filter(e => e.id === pageId)[0];
+
+    let index = 0;
+
+    recipe.flavors.forEach(recipeFlavor => {
+      let stashIndex = 0;
+
+      flavorStash.forEach(stashFlavor => {
+        if (recipeFlavor.id === stashFlavor.id) {
+          recipe.flavors[index].inStash = true;
+          return;
+        }
+
+        stashIndex++;
+        if (stashIndex === flavorStash.length) {
+          recipe.flavors[index].inStash = false;
+          return;
+        }
+      });
+
+      index++;
+    });
+
     this.setState(recipe);
   }
 
+  renderRatingButtons(ratingNum) {
+    const buttons = [];
+
+    let ratingKey = 1;
+
+    for (let i = 0; i < ratingNum; i++) {
+      buttons.push(
+        <Button
+          className="rating-button button--recipe"
+          key={i + 1}
+          onClick={() => this.handleRatingClick(i + 1)}
+        >
+          <span>
+            <FontAwesomeIcon icon={['fas', 'star']} />
+          </span>
+        </Button>
+      );
+      ratingKey++;
+    }
+
+    for (let a = ratingNum; a < 5; a++) {
+      buttons.push(
+        <Button
+          className="rating-button button--recipe"
+          key={a + 1}
+          onClick={() => this.handleRatingClick(a + 1)}
+        >
+          <span>
+            <FontAwesomeIcon icon={['far', 'star']} />
+          </span>
+        </Button>
+      );
+      ratingKey++;
+    }
+
+    if (ratingKey === 6) {
```

Comment:
Does this return undefined if ratingKey is not 6 at the end?

Reply:
I think I did this because I was running into issues with all 5 rating buttons rendering, so I added this to basically loop through and say once the loop is done, render the button group. Changed this as well though.